pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -127,6 +133,27 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 		if cephUsername, ok := secrets.Data[CephUsernameKey]; ok {
 			clusterInfo.CephCred.Username = string(cephUsername)
 			clusterInfo.CephCred.Secret = string(secrets.Data[CephUserSecretKey])
+			// for upgrade cluster get rookoperator keyrings
+			if string(cephUsername) == "client.admin" {
+
+				operatorCred, err := genAdminKeyring(clusterdContext, namespace)
```

Comment:
AuthGetOrCreateKey() should replace genAdminKeyring(). So you can remove this line, then use the results of AuthGetOrCreateKey().

Reply:
I'm also not confidence the config files have been generated under /var/lib/rook at this point when the operator is just starting, so if this fails it may be because the config needs to be generated.

Reply:
with only this `AuthGetOrCreateKey` called there was no config created and to create the config I'm using `genAdminKeyring` and regarding using `AuthGetOrCreateKey` key both `AuthGetOrCreateKey` and `genAdminKeyring` should return the same key for a single user